ui/src/components/Header/Menu.tsx

```diff
@@ -0,0 +1,62 @@
+import { MoreVert } from '@mui/icons-material';
+import { IconButton, Menu, MenuItem } from '@mui/material';
+import React, { useState } from 'react';
+import { UpdateDialog } from '../UpdateDialog';
+
+const ITEM_HEIGHT =80;
+
+export const LongMenu = () => {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [openModal, setOpenModal] = useState<boolean>(false);
+
+  const open = Boolean(anchorEl);
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleUpdateClick = () => {
+    handleClose();
+    setOpenModal(true);
+  };
+
+  return (
+    <div>
+      {openModal && <UpdateDialog open={openModal} onClose={() => setOpenModal(false)} />}
```

Comment:
seems like the conditional rendering is superfluous given the `open={openModal}` ...